### PR TITLE
MapRouletteUploadCommand: Handle gzipped files

### PR DESCRIPTION
**Pull request by nachtm**
### Description:

Previously, log files that were .gzipped were skipped by MapRouletteUploadCommand. Users that wanted to upload zipped files to MapRoulette would have to manually unzip them, which could be tedious. 

This PR gives MapRouletteUploadCommand the capability to handle `.log.gz` files on its own.

### Potential Impact:

None -- just new capabilities.

### Unit Test Approach:

Tried uploading gzipped files and non-zipped files.

### Test Results:

Given the following directory,
image
 
running with the following params
image

successfully shows up in MapRoulette, with all tasks present:
image

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java [hunk 1]
```diff
@@ -81,14 +88,68 @@ protected void execute(final CommandMap commandMap,
...
                 }
-            }
+            });
         });
     }
 
+    /**
+     * An enum containing the different types of input files that we can handle.
+     */
+    private enum HandledFileType
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java]: Something like `OutputFileType` would be more clear here

File: src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java [hunk 2]
```diff
@@ -81,14 +88,68 @@ protected void execute(final CommandMap commandMap,
...
+    {
+        LOG,
+        ZIPPED_LOG
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java]: `COMPRESSED_LOG`

File: src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java [hunk 3]
```diff
@@ -81,14 +88,68 @@ protected void execute(final CommandMap commandMap,
...
+
+    /**
+     * Determine whether or not this file is something we can handle, and classify it accordingly.
+     * 
+     * @param logFile
+     *            any file
+     * @return if this file is something this command can handle, the appropriate HandledFileType
+     *         enum value; otherwise, an empty optional.
+     */
+    private Optional<HandledFileType> maybeGetHandledType(final File logFile)
```
Comment (danielduhh) [inline src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java]: We should avoid words like `maybe` in function names 😆 . How about `getOutputFileType`?

**Comment by nachtm** [comment]
Thanks for the suggestions @danielduhh! As far as testing goes, I'm aware and thinking about the best way to test the bits I added. From what I can see, testing what's currently exposed would require uploading to Map Roulette. I'm open to suggestions on how to do this properly

**Comment by nachtm** [comment]
I've figured out a decent solution, but it relies on [TestMapRouletteConnection](https://github.com/osmlab/atlas-checks/pull/137/files#diff-a419b3c17abc6fe22df9eec29bf679a6) from #137. So, I'm going to close this until that gets pulled in and I can use it here.

**Comment by nachtm** [comment]
Since #137 got merged, I'm re-opening this!

**Comment by danielduhh** [comment]
@nachtm Is there anyway you can avoid committing the binary file ?

**Comment by nachtm** [comment]
@danielduhh Should be better now

**Pull request merged**